crates/hyperqueue/src/dashboard/ui/screens/home/chart_section.rs

```diff
@@ -0,0 +1,147 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+use tako::messages::gateway::CollectedOverview;
+
+use crate::dashboard::ui::terminal::DashboardFrame;
+use tui::layout::Rect;
+use tui::style::{Color, Modifier, Style};
+use tui::symbols;
+use tui::text::Span;
+use tui::widgets::{Axis, Block, Borders, Chart, Dataset};
+
+/// A new WorkerTimeState is created everytime there is a #worker change event
+#[derive(Clone)]
+struct WorkerStateAtTime {
+    /// The time at which the event being stored happened, wrt. the reference
+    pub at_time: u128,
+    pub event: TimeEventType,
+}
+
+#[derive(Clone)]
+pub enum TimeEventType {
+    /// Current number of workers.
+    WorkerCountUpdate(u32),
+    /// The average cpu utilization of the worker.
+    AvgCpuUsageUpdate(f32),
+}
+
+#[derive(Clone)]
+pub struct ClusterOverviewChart {
+    /// The state of the worker at different times.
+    timed_info_vec: Vec<WorkerStateAtTime>,
+    /// Used to set the min bound of the chart's x-axis
+    time_reference: u128,
+    /// Used to set the max bound of the chart's y-axis
+    max_observed_workers: u64,
+    /// Current number of workers connected
+    current_conn_workers: u64,
+}
+
+impl ClusterOverviewChart {
```

Comment:
The chart currently behaves in a quite weird way, because it start at the "beginning of time" and then it will just shrink forever to show the actual state. I think that instead we should do the following:
- Store the complete history of X and Y axis values received from the server (if this grows too large, we can instead use some rotating buffer, but let's ignore that for now).
- When drawing the chart, take some fixed amount of data from the event log (e.g. last 5 minutes) and only draw this "view" of the data. Later we can add moving in history, so the amount of history and its "location" will be able to be changed using e.g. keybinds.
- Compute the maximum of the Y axis based only on the subset of the data that is currently being drawn.
- Do not store `time_reference`. Use "current time" - "view size" (e.g. 5 minutes) as the reference value.